Backend/profile/profile.go

```diff
@@ -154,22 +162,143 @@ func FindUserByName(firstName string, lastName string) []models.User {
 	return returnList
 }
 
+
+func EditName(userID int, newFirst string, newLast string) error{
+
+	// Some checks included here to decide on editing first name, last name, or both. 
+	if len(newLast) > 0 && len(newFirst) > 0{
+
+		query := "UPDATE users SET first_name = ?, last_name = ? WHERE id = ?"
+		_, err := db.Exec(query,newFirst,newLast,userID)
+
+		if err != nil {
+			return err
+		}
+
+	}else if len(newLast) > 0{
+
+		query := "UPDATE users SET last_name = ? WHERE id = ?"
+		_, err := db.Exec(query,newLast,userID)
+
+		if err != nil {
+			return err
+		}
+
+	}else{
+
+		query := "UPDATE users SET first_name = ? WHERE id = ?"
+		_, err := db.Exec(query,newFirst,userID)
+
+		if err != nil {
+	
+		}
+
+	}
+
+	return nil
+}
+
+func EditBio(userID int, newBio string) error{
+
+	if len(newBio) > 0{
+
+		query := "UPDATE users SET bio = ? WHERE id = ?"
+		_, err := db.Exec(query,newBio,userID)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func EditUsername(userID int, newUsername string) error{
+
+	if len(newUsername) > 0{
+
+		query := "UPDATE users SET username = ? WHERE id = ?"
+		_, err := db.Exec(query,newUsername,userID)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func DeleteUser(userID int) error{
+
+	var username string
+	//Beginning transactions for the database, so they can be rolled back if an error occurs midway.
+	txn, err := db.Begin()
+
+	if err != nil {
+		return err
+	}
+
+	// deferring the function to either commit the transactions, or roll them back depending on if an error is thrown. 
+    defer func() {
+        if err != nil {
+            txn.Rollback()
+        } else {
+            err = txn.Commit()
+        }
+    }()
+
+	// getting username to delete from Friends table
+	err = txn.QueryRow("SELECT username FROM users WHERE id = ?",userID).Scan(&username)
+
+	// removing from reactions table
+	_, err = txn.Exec("DELETE FROM reactions WHERE user_id = ?",userID)
+	if err != nil {
+		return err
+	}
+
+	// removing from replies table
+	_, err = txn.Exec("DELETE FROM replies WHERE user_id = ?",userID)
+
+	if err != nil {
+		return err
+	}
+
+	// removing from posts table
+	_, err = txn.Exec("DELETE FROM posts WHERE user_id = ?",userID)
+
+	if err != nil {
+		return err
+	}
+
+	// removing from friends table
+	_, err = txn.Exec("DELETE FROM friends WHERE user_id = ?",username)
+
+	if err != nil {
+		return err
+	}
+
+	// Deleting the user
+	_, err = txn.Exec("DELETE FROM users WHERE id = ?",userID)
+
+	if err != nil {
+		return err
+	}
+
+
+	return nil
+
+}
```

Comment:
just change references from WHERE id = ? to WHERE username = ? and change the parameters